object/check.go

```diff
@@ -371,62 +387,62 @@ func (err *CheckUserPasswordError) RealError() error {
 	return err.err
 }
 
-func CheckUserPassword(ctx context.Context, organization string, username string, password string, lang string, options ...bool) (*User, error) {
+func CheckUserPassword(ctx context.Context, organization string, username string, password string, options CheckUserPasswordOptions) (*User, error) {
 	enableCaptcha := false
 	isSigninViaLdap := false
 	isPasswordWithLdapEnabled := false
-	if len(options) > 0 {
-		enableCaptcha = options[0]
-		isSigninViaLdap = options[1]
-		isPasswordWithLdapEnabled = options[2]
-	}
+	
+	enableCaptcha = options.EnableCaptcha
+	isSigninViaLdap = options.IsSigninViaLdap
+	isPasswordWithLdapEnabled = options.IsPasswordWithLdapEnabled
+	
 	user, err := GetUserByFields(organization, username)
 	if err != nil {
 		return nil, err
 	}
 
 	if user == nil || user.IsDeleted {
-		return nil, fmt.Errorf(i18n.Translate(lang, "general:The user: %s doesn't exist"), util.GetId(organization, username))
+		return nil, fmt.Errorf(i18n.Translate(options.Lang, "general:The user: %s doesn't exist"), util.GetId(organization, username))
```

Comment:
I think we should transfer Lang in context, because it's request scoped data.

Reply:
let fix in another issue